WordPress/openverse-api
# Add request id to logs

## Fixes


Part of #775  by @sarayourfriend 

## Description


Adds request IDs to all logs so that we can trace them easily throughout a request without having to rebuild it ourselves (we can simply query for the request ID and then follow a request's logs in their entirety).

I want to split out this part from https://github.com/WordPress/openverse-api/pull/776 because it applies cleanly without interrupting the revert of https://github.com/WordPress/openverse-api/pull/741 that @dhruvkb is going to work on. Getting request IDs so that we can trace logs through requests is useful at any point that we deploy next, even if we don't get in new diagnostic logging.

Initially I was going to write the filter and middleware myself but [this library](https://github.com/dabapps/django-log-request-id) appears well enough maintained and is simple enough that I am not worried about adding it. It also has apparently figured out some problems and edge cases that we would inevitably run into if we hand-spun this functionality.

## Testing Instructions

CI should pass. Checkout the branch locally and run the app and make a request (like to `/v1/images?q=waves`) and confirm you see something like this in your logs (noting the request id):

```
openverse-api-web-1               | [2022-06-29 05:33:18,289 - catalog.api.utils.validate_images - 103][INFO] [7d3d1f0b82334242a352eb0e4361283e] Validated images in 1.8959648609161377 
openverse-api-web-1               | [2022-06-29 05:33:18,328 - log_request_id.middleware -  47][INFO] [7d3d1f0b82334242a352eb0e4361283e] method=GET path=/v1/images/ status=200
```

Here, `7d3d1f0b82334242a352eb0e4361283e` is the request id.

## Checklist

- [N/A] I added or updated tests for the changes I made (if applicable).

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
